Slim/Control/Queries.pm

```diff
@@ -282,6 +282,7 @@ sub albumsQuery {
 	my $work	  = $request->getParam('work_id');
 	my $composerID    = $request->getParam('composer_id');
 	my $fromSearch    = $request->getParam('from_search');
+	my $allWorks      = $request->getParam('all_works');
```

Comment:
Now I'm confused... Isn't this what the `works` query would be for? Actually: could you please update the CLI documentation with the `works` query, so I understand what it is for vs. the albums query with the `work_id`?

Reply:
Yes, sorry about not doing the documentation yet - there always seems something else to do first! I'll get on to it, I promise.

The works query would return all instances of a work, so if we used that for the Random Mix,  all versions of the work in the library would be added to the random playlist as a block, which is not what's required. An individual performance of a work is identified by album, work and grouping.

So as when drilling down from a work to its individual performances, I use the albums query, but obviously without passing in a work_id, we need something else to tell it to join to and group by work (and grouping), prepare the favorites URL properly, etc.

Reply:
`works` would return works, no matter the recording, and `albums work_id:123` would return the actual recording of a work?

Reply:
`albums work_id:123` will return a row for each album which contains the work and satisfies any other input criteria (actually, if the work exists on an album multiple times, distinguished by the grouping tag, it will return multiple rows for that album, as well). So you end up with a row for each performance of the work that you have.

`works` returns a single row for each work, ready to drill down to individual performances using `albums`.